test/integration/features/addon-registry-route.feature

```diff
@@ -0,0 +1,50 @@
+@addon-registry-route @addon
+Feature: registry-route add-on
+registry-route addon exposes the route for the OpenShift integrated registry,
+so that user can able to push their container images directly to registry and deploy it.
+
+  Scenario: As a user, I can enable the registry-route add-on
+    Given user starts shell instance on host machine
+     When executing "minishift addons enable registry-route" succeeds
+     Then stdout should contain "Add-on 'registry-route' enabled"
+
+  Scenario: As a user, I can start Minishift with registry-route addon enabled
+    Given Minishift has state "Does Not Exist"
+      And image caching is disabled
+     When executing "minishift start --iso-url centos" succeeds
+     Then Minishift should have state "Running"
+      And stdout should contain
+      """
+      Add-on 'registry-route' created docker-registry route.
+      """
+
+  Scenario: As a user, I can login to OpenShift integrated docker registry
+    Given Minishift has state "Running"
+      And executing "minishift docker-env" in host shell succeeds
+      And evaluating stdout of the previous command in host shell
+      And executing "minishift oc-env" in host shell succeeds
+      And evaluating stdout of the previous command in host shell
+     Then with up to "20" retries with wait period of "3" seconds container image "origin-docker-registry" should be "running"
+      And executing "curl -k --retry 60 --retry-delay 1 --retry-connrefused https://docker-registry-default.$(minishift ip).nip.io" in host shell succeeds
```

Comment:
What this line is doing?
I think we need a step definition here similar to "^executing "oc (.*)" retrying (\d+) times with wait period of (\d+) seconds$"
as

```
^executing "(.*)" retrying (\d+) times with wait period of (\d+) seconds$
```

Reply:
@budhrg https://github.com/minishift/minishift/pull/2158#issuecomment-376046097

Reply:
> ^executing "(.*)" retrying (\d+) times with wait period of (\d+) seconds$

+1. Readability would be nice

Reply:
Done